src/anomalib/metrics/per_image/pimo_numpy.py

```diff
@@ -303,7 +296,7 @@ def aupimo_scores(
             "Try increasing `num_threshs`.",
         )
 
-    aucs: ndarray = np.trapezoid(per_image_tprs_bounded, x=shared_fpr_bounded_log, axis=1)
+    aucs: ndarray = np.trapz(per_image_tprs_bounded, x=shared_fpr_bounded_log, axis=1)  # noqa: NPY201 deprecated in Numpy 2.0
```

Comment:
imgaug does not support numpy 2.x, and np.trapezoid was introduced in numpy 2.x. Once imgaug is removed we can look into reverting this.

Reply:
Can we do something like this to avoid disabling ruff, and be more clear?

```python
trapezoid = np.trapezoid if np.__version__ >= "2.0.0" else np.trapz
aucs: ndarray = trapezoid(per_image_tprs_bounded, x=shared_fpr_bounded_log, axis=1)
```

Reply:
Ruff still throws an error. And, we won't need this once we switch everything to torch

Reply:
fair enough